File: sdk/php/src/Resources/Pmg.php
<?php

namespace HiTechCloud\SDK\Resources;

class Pmg extends BaseResource
{
    /**
     * Get config
     */
    public function getConfig(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/htcpmg/config');
    }

    /**
     * Add domain
     */
    public function addDomain(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/htcpmg/domains', $data);
    }

    /**
     * Set transport
     */
    public function setTransport(int $serviceId, array $data): array
    {
        return $this->http->post('/api/service/' . $serviceId . '/htcpmg/transport', $data);
    }

    /**
     * Get stats
     */
    public function getStats(int $serviceId): array
    {
        return $this->http->get('/api/service/' . $serviceId . '/htcpmg/stats');
    }

}

File: sdk/csharp/Resources/Pmg.cs
using System.Collections.Generic;
using System.Threading.Tasks;

namespace HiTechCloud.SDK.Resources;

public class Pmg : BaseResource
{
    public Pmg(HttpClientWrapper http) : base(http) { }

    public async Task<Dictionary<string, object?>> ConfigAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/htcpmg/config", id));
    }

    public async Task<Dictionary<string, object?>> AddDomainAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/htcpmg/domains", id), data);
    }

    public async Task<Dictionary<string, object?>> AddTransportAsync(int id, Dictionary<string, object> data)
    {
        return await Http.PostAsync(string.Format("/api/service/{id}/htcpmg/transport", id), data);
    }

    public async Task<Dictionary<string, object?>> StatsAsync(int id)
    {
        return await Http.GetAsync(string.Format("/api/service/{id}/htcpmg/stats", id));
    }
}
